trie/triedb/pathdb/errors.go

```diff
@@ -25,9 +25,13 @@ import (
 )
 
 var (
-	// errSnapshotReadOnly is returned if the database is opened in read only mode
-	// and mutation is requested.
-	errSnapshotReadOnly = errors.New("read only")
+	// errDatabaseReadOnly is returned if the database is opened in read only mode
+	// to prevent any mutation.
+	errDatabaseReadOnly = errors.New("read only")
+
+	// errDatabaseDisabled is returned if database is disabled due to an ongoing
+	// state sync process.
+	errDatabaseDisabled = errors.New("disabled")
```

Comment:
Nitpick. Perhaps don't say "disabled" because it's gonna be very hard for someone hitting it to figure out what's happening. Could we perhaps have "waiting for sync" / `errDatabaseWaitSync`? It's a lot more specific but I feel it would be clearer for anyone looking at the code later.